src/common/hwio_pindef_MINI.h

```diff
@@ -1,46 +1,124 @@
 //----------------------------------------------------------------------------//
 // pin names
-#ifndef _HWIO_PINDEF_H
-#define _HWIO_PINDEF_H
-
-#define PIN_BEEPER         TPA0
-#define PIN_HW_IDENTIFY    TPA3
-#define PIN_TEMP_BED       TPA4
-#define PIN_THERM2         TPA5
-#define PIN_TEMP_HEATBREAK TPA6
-#define PIN_Z_MIN          TPA8
-
-#define PIN_HEATER_BED TPB0
-#define PIN_HEATER_0   TPB1
-
-#define PIN_TEMP_0 TPC0
-
-#define PIN_X_DIR    TPD0
-#define PIN_X_STEP   TPD1
-#define PIN_X_ENABLE TPD3
-#define PIN_X_DIAG   TPE2
-
-#define PIN_Y_DIR    TPD12
-#define PIN_Y_STEP   TPD13
-#define PIN_Y_ENABLE TPD14
-#define PIN_Y_DIAG   TPE1
-
-#define PIN_Z_DIR    TPD15
-#define PIN_Z_STEP   TPD4
-#define PIN_Z_ENABLE TPD2
-#define PIN_Z_DIAG   TPE3
-
-#define PIN_E_DIR    TPD8
-#define PIN_E_STEP   TPD9
-#define PIN_E_ENABLE TPD10
-#define PIN_E_DIAG   TPA15
-
-#define PIN_FAN1    TPE9
-#define PIN_FAN     TPE11
-#define PIN_BTN_ENC TPE12
-#define PIN_BTN_EN1 TPE13
-#define PIN_BTN_EN2 TPE15
+#pragma once
 
-#define PIN_FSENSOR TPB4
+#include "Pin.hpp"
+
+#define MARLIN_PORT_PIN_HW_IDENTIFY    MARLIN_PORT_PIN(MARLIN_PORT_A, MARLIN_PIN_3) //ADC, unused
+#define MARLIN_PORT_PIN_TEMP_BED       MARLIN_PORT_PIN(MARLIN_PORT_A, MARLIN_PIN_4) //ADC
+#define MARLIN_PORT_PIN_THERM2         MARLIN_PORT_PIN(MARLIN_PORT_A, MARLIN_PIN_5) //ADC
+#define MARLIN_PORT_PIN_TEMP_HEATBREAK MARLIN_PORT_PIN(MARLIN_PORT_A, MARLIN_PIN_6) //ADC
+#define MARLIN_PORT_PIN_TEMP_0         MARLIN_PORT_PIN(MARLIN_PORT_C, MARLIN_PIN_0) //ADC
+
+#define MARLIN_PORT_Z_MIN     MARLIN_PORT_A
+#define MARLIN_PIN_Z_MIN      MARLIN_PIN_8
+#define MARLIN_PORT_PIN_Z_MIN MARLIN_PORT_PIN(MARLIN_PORT_Z_MIN, MARLIN_PIN_Z_MIN)
+extern InputPin zMin;
+
+#define MARLIN_PORT_X_DIAG     MARLIN_PORT_E
+#define MARLIN_PIN_X_DIAG      MARLIN_PIN_2
+#define MARLIN_PORT_PIN_X_DIAG MARLIN_PORT_PIN(MARLIN_PORT_X_DIAG, MARLIN_PIN_X_DIAG)
+extern InputPin xDiag;
+
+#define MARLIN_PORT_Y_DIAG     MARLIN_PORT_E
+#define MARLIN_PIN_Y_DIAG      MARLIN_PIN_1
+#define MARLIN_PORT_PIN_Y_DIAG MARLIN_PORT_PIN(MARLIN_PORT_Y_DIAG, MARLIN_PIN_Y_DIAG)
```

Comment:
/for discussion/ Having the `MARLIN_PORT_PIN_<NAME>` with every pin definition seems a wee bit repetitive. What about something like this:

```C
#define MARLIN_PORT_<NAME> <VALUE>
#define MARLIN_PIN_NUMBER_<NAME> <VALUE>
```
and then having a global `MARLIN_PIN(name)` macro, which would expand to `MARLIN_PORT_PIN(MARLIN_PORT_<NAME>, MARLIN_PIN_NR_<NAME>)`, changing the usage to `MARLIN_PIN(Y_DIAG)` in the case above.

Reply:
done